upstairs/src/client.rs

```diff
@@ -2916,108 +2916,51 @@ pub(crate) fn validate_encrypted_read_response(
             error!(log, "got empty block context with non-blank block");
             return Err(CrucibleError::MissingBlockContext);
         }
-    }
-
-    let mut valid_hash = None;
-    let mut successful_decryption = false;
-
-    // Attempt decryption with each encryption context, and fail if all
-    // do not work. The most recent encryption context will most likely
-    // be the correct one so start there.
-    for ctx in block_contexts.iter().rev() {
-        let block_encryption_ctx =
-            if let Some(block_encryption_ctx) = &ctx.encryption_context {
-                block_encryption_ctx
-            } else {
-                // this block context is missing an encryption context!
-                // continue to see if another block context has a valid one.
-                //
-                // XXX should this be an error instead?
-                continue;
-            };
-
-        // Validate integrity hash before decryption
-        let computed_hash = integrity_hash(&[
-            &block_encryption_ctx.nonce[..],
-            &block_encryption_ctx.tag[..],
+    };
+
+    let Some(block_encryption_ctx) = &context.encryption_context else {
+        // this block context is missing an encryption context!
+        // XXX should this be an error instead?
```

Comment:
Missed XXX cleanup?

Reply:
[Pre-existing condition 😉 ](https://github.com/oxidecomputer/crucible/blob/49be9264d56cf55bc3d9e2e4ec7d58540e18fa82/upstairs/src/client.rs#L2935)

Reply:
🤔 the next line is
```
return Err(CrucibleError::DecryptionError);
```

Unless the comment is referring to a new error type.

Reply:
lol, sorry for the duplicate review comment.